### PR TITLE
feat(ingest):  mariadb lineage 

### PR DESCRIPTION
**Pull request by sebastianpenning**
I wrote a draft python code for getting the lineage from a mariadb database based on foreign key relationships retrieved with SQLAlchemy. This is useful to see the relationships inside the database, and how they could have potential impact downstream in data pipelines / upstream in production applications, Due to limited experience with OOP, I would need help to convert this into working code with the rest of the repository.  

In the future, I think we can also get column based lineage since SQLAlchemy also allows you to see which columns are referenced by the foreign keys.  

This PR started from a discussion in Slack: https://datahubspace.slack.com/archives/CUMUWQU66/p1671024198126679

### CODE REVIEW

**Comment by xiphl** [comment]
This has the effect of drawing up the relationship diagram between tables, rather than how the table are transformed into another table, does it not?

**Comment by sebastianpenning** [comment]
@xiphl Exactly, so maybe this should be called more like relational lineage instead of just lineage. I developed this because mariadb is used as a production db for our in-house developed applications. The data from this db is then transformed in a data pipeline. Therefore, it made sense to have a relational lineage overview for both the software developers, to see what there impact would be on the in-house applications when altering tables / views in the db, and data developers for the same purposes + making it easier to apply transformations / business logic. The functionality is like that of Schemaspy

File: .github/workflows/docker-unified.yml [hunk 1]
```diff
@@ -23,14 +23,14 @@ concurrency:
...
 env:
-  DATAHUB_GMS_IMAGE: "linkedin/datahub-gms"
-  DATAHUB_FRONTEND_IMAGE: "linkedin/datahub-frontend-react"
-  DATAHUB_MAE_CONSUMER_IMAGE: "linkedin/datahub-mae-consumer"
-  DATAHUB_MCE_CONSUMER_IMAGE: "linkedin/datahub-mce-consumer"
-  DATAHUB_KAFKA_SETUP_IMAGE: "linkedin/datahub-kafka-setup"
-  DATAHUB_ELASTIC_SETUP_IMAGE: "linkedin/datahub-elasticsearch-setup"
-  DATAHUB_MYSQL_SETUP_IMAGE: "acryldata/datahub-mysql-setup"
-  DATAHUB_UPGRADE_IMAGE: "acryldata/datahub-upgrade"
+  DATAHUB_GMS_IMAGE: "spenning/datahub-gms"
```
Comment (jjoyce0510) [inline .github/workflows/docker-unified.yml:26]: Please revert

File: docker/quickstart/docker-compose.quickstart.yml [hunk 2]
```diff
@@ -148,7 +148,7 @@ services:
     - KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181
     - KAFKA_BOOTSTRAP_SERVER=broker:29092
     hostname: kafka-setup
-    image: ${DATAHUB_KAFKA_SETUP_IMAGE:-linkedin/datahub-kafka-setup}:${DATAHUB_VERSION:-head}
+    image: ${DATAHUB_KAFKA_SETUP_IMAGE:-spenning/datahub-kafka-setup}:${DATAHUB_VERSION:-head}
```
Comment (jjoyce0510) [inline docker/quickstart/docker-compose.quickstart.yml:151]: Let's revert

**Review by jjoyce0510 — COMMENTED** [review]
It looks like we are constructing lineage from foreign key references. 
Typically, we use lineage to denote a _derivation_ relationship, wherein a table is derived by applying computation to one or more upstream / input tables. 

We do have a feature request open for supporting [ERD modeling ](https://feature-requests.datahubproject.io/p/ui-support-for-erd)in the catalog, but have not yet formally decided to take it on. 

For that reason, I'm going to close this PR until we do intend to support that feature, at which point we can reconsider. For now, we won't be able to accept.
 
Thank you so much for being an active member of the DataHub Community. Please feel free to reach out with any questions or concerns! 

Cheers
John

**Pull request closed without merging**